llvm/lib/Target/PowerPC/PPCTargetTransformInfo.cpp

```diff
@@ -885,6 +885,19 @@ PPCTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
   return BaseT::getIntrinsicInstrCost(ICA, CostKind);
 }
 
+bool PPCTTIImpl::areInlineCompatible(const Function *Caller,
+                                     const Function *Callee) const {
+  // Allow inlining only when the Callee has a subset of the Caller's features.
+  const TargetMachine &TM = getTLI()->getTargetMachine();
+
+  const FeatureBitset &CallerBits =
+      TM.getSubtargetImpl(*Caller)->getFeatureBits();
+  const FeatureBitset &CalleeBits =
+      TM.getSubtargetImpl(*Callee)->getFeatureBits();
+
+  return (CallerBits & CalleeBits) == CalleeBits;
```

Comment:
Looks like the X86 equivalent function also checks for ABI compatibility.  Wonder if that is also needed for PPC, for instance for vec-extabi.

Reply:
+1
And except for the functionality issue, do we need to exclude the PPC features that should not impact the inline optimization, like fusion related features `fuse-store`, `fuse-zeromove` and other optimization related features like `fast-MFLR`, `modern-aix-as`, `ppc-postra-sched`... Inline optimization may introduce bigger improvement than breaking these optimization related features?